storage/src/test/java/org/apache/uniffle/storage/common/DefaultStorageMediaProviderTest.java

```diff
@@ -40,7 +40,7 @@ public void testStorageProvider() {
         StorageMedia.OBJECT_STORE, provider.getStorageMediaFor("cos://bucket-name/b/path"));
 
     // by default, the local file should report as HDD
-    assertEquals(StorageMedia.HDD, provider.getStorageMediaFor("/path/to/base/dir"));
+    assertEquals(StorageMedia.HDD, provider.getStorageMediaFor("path/to/base/dir"));
     assertEquals(StorageMedia.HDD, provider.getStorageMediaFor("file:///path/to/a/dir"));
```

Comment:
I think the result of this line is correct, but the underlying logic is wrong. Currently, if the path contains `file://`, `DefaultStorageMediaProvider#getFileStore` will return null. I think the follow line should be changed to 
`URI uri = new URI(baseDir);`. 
https://github.com/apache/uniffle/blob/76a4dcf76b41529dfda36569140c48c3f0554722/storage/src/main/java/org/apache/uniffle/storage/common/DefaultStorageMediaProvider.java#L65

And then this line should be changed as the follow.
```suggestion
    assertEquals(provider.getStorageMediaFor("/") provider.getStorageMediaFor("file:///path/to/a/dir"));
```
WDYT? @advancedxy @asalamon74

Reply:
If I change the

```
File baseFile = new File(baseDir); 
```

line to 

```
URI uri = new URI(baseDir);
```

I've got `java.lang.IllegalArgumentException: Missing scheme` error if we don't use the scheme.

So I think we need both lines. I modified the code, first I use `new File` and if it returns null and the scheme is `file` I'll use the URI approach.